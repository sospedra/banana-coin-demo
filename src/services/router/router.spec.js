/* global expect, describe, it */
const { promisify, createRouter } = require('./')

jest.mock('express', () => ({
  Router: jest.fn(() => ({ get: jest.fn() }))
}))

describe('service router', () => {
  it('should promisify a callback with req, res, next', async () => {
    const callback = jest.fn()
    const args = [1, 2, 3]
    const promise = promisify(callback)(...args)

    await promise

    expect(promise).toBeInstanceOf(Promise)
    expect(callback).toHaveBeenCalledWith(...args)
  })

  it('should create a router given some routes', () => {
    const route = () => {}
    route.verb = 'get'
    route.path = '/'

    expect(createRouter({ route })).toMatchSnapshot()
  })
})
