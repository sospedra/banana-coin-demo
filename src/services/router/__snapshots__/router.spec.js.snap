// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service router should create a router given some routes 1`] = `
Object {
  "get": [MockFunction] {
    "calls": Array [
      Array [
        "/",
        [Function],
      ],
    ],
  },
}
`;
